apps/opik-backend/src/main/java/com/comet/opik/domain/ProjectMetricsDAO.java

```diff
@@ -25,55 +34,161 @@
+    @Override
+    public Mono<List<Entry>> getFeedbackScores(@NonNull UUID projectId, @NonNull ProjectMetricRequest request) {
+        return template.nonTransaction(connection -> getMetric(projectId, request, connection,
+                GET_FEEDBACK_SCORES, "feedbackScores")
+                .flatMapMany(result -> rowToDataPoint(
+                        result,
+                        request,
+                        row -> row.get("name", String.class),
+                        row -> row.get("value", BigDecimal.class)))
+                .collectList());
     }
 
-    private Mono<? extends Result> getTracesCountForProject(
-            UUID projectId, ProjectMetricRequest request, Connection connection) {
-        var template = new ST(GET_TRACE_COUNT);
+    private Mono<? extends Result> getMetric(
+            @NonNull UUID projectId, @NonNull ProjectMetricRequest request, @NonNull Connection connection,
+            @NonNull String query, @NonNull String segmentName) {
+        var template = new ST(query)
+                .add("convert_interval", intervalToSql(request.interval()))
+                .add("is_weekly", request.interval() == TimeInterval.WEEKLY);
         var statement = connection.createStatement(template.render())
                 .bind("project_id", projectId)
                 .bind("start_time", request.intervalStart().toString())
-                .bind("end_time", request.intervalEnd().toString());
+                .bind("end_time", request.intervalEnd().toString())
+                .bind("backfill_start_time", request.intervalStart() == Instant.EPOCH ?
+                        getAllTimeBackfillStart().toString() :
+                        request.intervalStart().toString());
 
-        InstrumentAsyncUtils.Segment segment = startSegment("traceCount", "Clickhouse", "get");
+        InstrumentAsyncUtils.Segment segment = startSegment(segmentName, "Clickhouse", "get");
 
         return makeMonoContextAware(bindWorkspaceIdToMono(statement))
                 .doFinally(signalType -> endSegment(segment));
     }
 
-    private Publisher<DataPoint> mapToIntDataPoint(Result result) {
-        return result.map(((row, rowMetadata) -> DataPoint.builder()
-                .time(row.get("bucket", Instant.class))
-                .value(row.get("count", Integer.class))
+    private Publisher<Entry> rowToDataPoint(
+            @NonNull Result result,
+            @NonNull ProjectMetricRequest request,
+            @NonNull Function<Row, String> nameGetter,
+            @NonNull Function<Row, ? extends Number> valueGetter) {
+        return result.map(((row, rowMetadata) -> Entry.builder()
+                .name(nameGetter.apply(row))
+                .value(valueGetter.apply(row))
+                .time(extractBucket(request, row))
                 .build()));
     }
+
+    private static Instant extractBucket(@NonNull ProjectMetricRequest request, @NonNull Row row) {
+        if (request.interval() == TimeInterval.WEEKLY) {
```

Comment:
Unfortunately no. Please notice that when performing the weekly aggregation, Clickhouse returns a `LocalDate` instead of an `Instance`. This statement merely converts the `LocalDate` object to `Instant`.

Reply:
But using the template, you could check with an if and apply the conversion only of that type. The issue is adding statements like `ZoneId.of("UTC")`. We are concentrating this at the DB level and only using Instant in the application.